src/server/game/Scripting/ScriptMgr.cpp

```diff
@@ -1670,6 +1670,15 @@ bool ScriptMgr::OnAreaTrigger(Player* player, AreaTriggerEntry const* trigger, b
     return tmpscript->OnTrigger(player, trigger, entered);
 }
 
+bool ScriptMgr::OnAreaTriggerServer(Player* player, AreaTriggerTemplate const* triggerTemplate, bool entered)
+{
+    ASSERT(player);
+    ASSERT(triggerTemplate);
+
+    GET_SCRIPT_RET(AreaTriggerScript, triggerTemplate->ScriptId, tmpscript, false);
+    return tmpscript->OnTriggerServer(player, triggerTemplate, entered);
+}
+
```

Comment:
ScriptMgr changes are unneccessary, scripts can use existing `AreaTriggerAI` by just giving it the scriptname in db